File: app/src/main/java/com/iml1s/xmrigminer/data/model/MiningConfig.kt
package com.iml1s.xmrigminer.data.model

import kotlinx.serialization.Serializable

@Serializable
data class MiningConfig(
    val poolUrl: String = "pool.supportxmr.com:3333",
    val walletAddress: String = "",
    val workerName: String = "android",
    val threads: Int = Runtime.getRuntime().availableProcessors() - 1,
    val maxCpuUsage: Int = 75,
    val useTls: Boolean = true,
    val autoReconnect: Boolean = true,
    val mineWhenScreenOff: Boolean = false,
    val donateLevel: Int = 0,  // 強制 0%
    val customArgs: String = "",
    val retries: Int = 5,
    val retryPause: Int = 5,
    val printTime: Int = 60
) {
    fun toJson(): String {
        return """
        {
            "autosave": false,
            "cpu": {
                "enabled": true,
                "max-threads-hint": $maxCpuUsage,
                "priority": 1,
                "asm": true,
                "argon2-impl": "auto"
            },
            "pools": [
                {
                    "url": "$poolUrl",
                    "user": "$walletAddress",
                    "pass": "$workerName",
                    "keepalive": true,
                    "tls": $useTls
                }
            ],
            "donate-level": 0,
            "log-file": null,
            "print-time": $printTime,
            "health-print-time": $printTime,
            "retries": $retries,
            "retry-pause": $retryPause,
            "api": null,
            "http": null
        }
        """.trimIndent()
    }

    fun isValid(): Boolean {
        return walletAddress.isNotBlank() && 
               poolUrl.isNotBlank() &&
               threads > 0 &&
               maxCpuUsage in 10..100
    }
}

File: ios/XMRigMiner-iOS/Sources/Models/MiningConfig.swift
//
//  MiningConfig.swift
//  XMRigMiner-iOS
//
//  Mining configuration model
//

import Foundation

/// Mining pool configuration
struct PoolConfig: Codable {
    var url: String
    var user: String
    var pass: String
    var tls: Bool
    
    init(url: String = "pool.supportxmr.com:3333",
         user: String = "",
         pass: String = "x",
         tls: Bool = true) {
        self.url = url
        self.user = user
        self.pass = pass
        self.tls = tls
    }
}

/// Complete mining configuration
struct MiningConfig: Codable {
    var pool: PoolConfig
    var threads: Int
    var cpuPriority: Int
    var donateLevel: Int
    
    init(pool: PoolConfig = PoolConfig(),
         threads: Int = 0,
         cpuPriority: Int = 2,
         donateLevel: Int = 1) {
        self.pool = pool
        self.threads = threads
        self.cpuPriority = cpuPriority
        self.donateLevel = donateLevel
    }
    
    /// Convert to XMRig JSON config format
    func toJSON() -> String? {
        let config: [String: Any] = [
            "autosave": true,
            "cpu": [
                "enabled": true,
                "max-threads-hint": threads > 0 ? threads * 25 : 75
            ],
            "donate-level": donateLevel,
            "pools": [
                [
                    "url": pool.url,
                    "user": pool.user,
                    "pass": pool.pass,
                    "tls": pool.tls,
                    "keepalive": true
                ]
            ]
        ]
        
        guard let data = try? JSONSerialization.data(withJSONObject: config, options: .prettyPrinted),
              let jsonString = String(data: data, encoding: .utf8) else {
            return nil
        }
        
        return jsonString
    }
}

// MARK: - Preset Pools

extension PoolConfig {
    static let supportXMR = PoolConfig(url: "pool.supportxmr.com:3333", tls: true)
    static let moneroOcean = PoolConfig(url: "gulf.moneroocean.stream:10128", tls: true)
    static let hashVault = PoolConfig(url: "pool.hashvault.pro:3333", tls: true)
    static let twoMiners = PoolConfig(url: "xmr.2miners.com:2222", tls: false)
}
